CorsixTH/Lua/dialogs/furnish_corridor.lua

```diff
@@ -139,11 +147,14 @@ function UIFurnishCorridor:purchaseItem(index, quantity)
   elseif quantity > 99 then
     quantity = 99
   end
-  quantity = quantity - o.qty
+
+  local bought = quantity - o.qty
   local hospital = self.ui.hospital
-  if hospital.balance >= self.total_price + quantity * hospital:getObjectBuildCost(o.object.id) or is_negative_quantity then
-    o.qty = o.qty + quantity
-    self.total_price = self.total_price + quantity * hospital:getObjectBuildCost(o.object.id)
+  local total_price = self.total_price + bought * hospital:getObjectBuildCost(o.object.id)
+
+  if hospital.balance >= total_price or is_negative_quantity then
+    o.qty = math.tointeger(o.qty + bought)
```

Comment:
This math function isn't supported by CorsixTH. We use the Lua 5.1 base
https://www.codecademy.com/resources/docs/lua/mathematical-library/tointeger

However, I'm not really sure how this number is not remaining an integer.

Reply:
My bad, the issue is that I have lua 5.4 installed, and the CMake configuration doesn't guarantee nor checks if the lua version is the 5.1, i think it would be better if we forced the download of the correct lib, or at least show some error